sdk/monitor/azure-monitor-query/v2_migration_guide.md

```diff
@@ -0,0 +1,333 @@
+# Guide to migrating from v1 to v2
+
+Version 2.0.0 of `azure-monitor-query` removes the metrics querying capabilities which are now handled by separate packages.
+This guide assists you in the migration from [azure-monitor-query](https://pypi.org/project/azure-monitor-query/) versions 1.x to 2.x, and outlines how to adapt your code to leverage the new packages for metrics querying.
+
+## Table of contents
+
+- [Quick migration decision tree](#quick-migration-decision-tree)
+- [Motivation for v2](#motivation-for-v2)
+- [Key changes](#key-changes)
+  - [Logs querying with LogsQueryClient](#logs-querying-with-logsqueryclient)
+  - [Metrics querying with MetricsClient](#metrics-querying-with-metricsclient)
+  - [Metrics operations with MetricsQueryClient](#metrics-operations-with-metricsqueryclient)
+    - [Querying metrics](#querying-metrics)
+    - [Listing metric definitions](#listing-metric-definitions)
+    - [Listing metric namespaces](#listing-metric-namespaces)
+- [Common migration tasks](#common-migration-tasks)
+
+## Quick migration decision tree
+
+**Which client are you currently using?**
+
+- **`LogsQueryClient`** → **No changes needed** - continue using `azure-monitor-query`
+- **`MetricsClient`** → **Simple migration** - switch to `azure-monitor-querymetrics` package (same API, different import)
+- **`MetricsQueryClient`** → **Choose your path:**
+  - For **metrics querying**: Use `MetricsClient` from `azure-monitor-querymetrics` (recommended - higher limits)
+  - For **full metrics management capabilities**: Use `MonitorManagementClient` from `azure-mgmt-monitor` (requires code changes)
+
+## Motivation for v2
+
+Historically, the Azure Monitor Query library was a monolithic package that included both log analytics and metrics querying capabilities across three different services/endpoints. While this may have seemed convenient, it introduced challenges for users. A single SDK package combining multiple services meant customers couldn’t upgrade one service independently of others. Any breaking change in one service forced updates across all services in the package, disrupting customer workflows.
```

Comment:
```suggestion
Historically, the Azure Monitor Query library was a monolithic package that included both Log Analytics and Metrics querying capabilities across 5 different services/endpoints. While this may have seemed convenient, it introduced challenges for users. A single package combining multiple services meant customers couldn’t upgrade one service independently of others. Any breaking change in one service forced updates across all services in the package, disrupting customer workflows.
```